packages/fuels-test-helpers/src/chains.rs

```diff
@@ -0,0 +1,87 @@
+use crate::{utility, Opcode, Transaction, REG_ONE};
+use fuel_types::AssetId;
+use fuels_core::constants::{
+    BASE_ASSET_ID, DEFAULT_BYTE_PRICE, DEFAULT_GAS_LIMIT, DEFAULT_GAS_PRICE, DEFAULT_MATURITY,
+    DEFAULT_SPENDABLE_COIN_AMOUNT,
+};
+use fuels_core::errors::Error;
+use fuels_core::tx::{Input, Output};
+use fuels_signers::provider::Provider;
+use fuels_signers::{LocalWallet, Signer};
+use rand::Rng;
+use std::time::Duration;
+use utility::retry_until;
+
+/// Can be used to generate `amount` number of blocks. A block is generated by
+/// submitting a no-op script for execution. Transactions are submitted one
+/// after the other. A new transaction won't be submitted until an increase of
+/// block height has been noticed.
+///
+/// # Arguments
+///
+/// * `wallet`: A `Wallet` that must contain a provider and spendable coins.
+/// * `amount`: By how much to increase the block height.
+///
+/// returns: Result<(), Error>
+pub async fn add_blocks(wallet: &LocalWallet, amount: usize) -> Result<(), Error> {
+    let provider = wallet.get_provider()?;
+    for _ in 0..amount {
+        let height_before_transaction = provider.latest_block_height().await?;
+
+        let transaction = no_op_signed_transaction(wallet).await?;
+
+        provider.send_transaction(&transaction).await?;
+
+        if !check_if_block_height_increased(provider, height_before_transaction).await? {
+            return Err(Error::InfrastructureError(format!("Couldn't confirm a block generation via no-op script -- the block height ({}) stayed the same!", height_before_transaction)));
+        }
+    }
+    Ok(())
+}
+
+async fn no_op_signed_transaction(wallet: &LocalWallet) -> Result<Transaction, Error> {
+    let inputs = wallet
+        .get_asset_inputs_for_amount(BASE_ASSET_ID, DEFAULT_SPENDABLE_COIN_AMOUNT, 0)
+        .await?;
+
+    let outputs = vec![Output::change(wallet.address(), 0, AssetId::default())];
+
+    let mut transaction = generate_no_op_script(inputs, outputs);
+
+    wallet.sign_transaction(&mut transaction).await?;
+
+    Ok(transaction)
+}
+
+async fn check_if_block_height_increased(
+    provider: &Provider,
+    height_to_compare_with: u64,
```

Comment:
`previous_height` is a more concise argument name

Reply:
I initially did `previous_height` then I thought... what does previous mean in this context? This function doesn't know about the loop it is used in.

But after your comment I see now that `previous` could work because we are asking 'has the block height increased'. Applying the change.